src/swap.ts

```diff
@@ -48,7 +55,7 @@ export class Swap {
    * This is only valid for ERC20 swaps.
    *
    * @param tryParams Controls at which stage the exception is thrown.
-   * @returns The hash of the transaction that was sent to the blockchain network.
+   * @returns The hash of the transaction that was sent to the blockchain network. Throws if ???
```

Comment:
Imho this can throw due to an error with cnd OR with the blockchain node . Same for the other LedgerActions

Reply:
I am sorry but this comment is confusing to me. I explicitly wrote in the description of the PR: "On functions that interacts with cnd and a ledger, cnd errors bubbles up via throws but ledger action are returned as result in case it returns an error."

So no, the aim would be for this function to return a `Promise<Result>` and not throw on blockchain errors, only throw on cnd errors.

Reply:
I wanted to finish your sentence as it seemed like you did not know what to write there.

Reply:
I see. The sentence is not finish because it is exactly the topic of this review. Do we throw everything (cnd and wallets errors), just cnd errors, nothing?

Reply:
Ah, got it. 
I would opt for throwing in all cases.